test/platform.test.ts

```diff
@@ -46,7 +46,7 @@ suite("Platform module", () => {
 
             checkDefaultPowerShellPath(
                 platformDetails,
-                platform.System32PowerShellPath);
+                platform.getAvailablePowerShellExes(platformDetails, undefined)[0].exePath);
```

Comment:
It seems brittle to rely on a specific index in the array returned by `getAvailablePowerShellExes()`.  Someone might change that method in a way that reorders the elements in the array - which would seem like a safe thing to do. But it would break this code.  I would filter this by the versionName field or if that is inconvenient, perhaps add a new field to the objects returned by `getAvailablePowerShellExes` with the info you want to filter on.

This also applies to lines 79 and 109.

Reply:
I completely agree, just hadnt thought of a better way to do it...I will look into those options

Reply:
How about something like:

```typescript
const psExePaths = platform.getAvailablePowerShellExes(platformDetails).reduce((acc, psExe) => {
    acc[psExe.versionName] = psExe
}, {});

...

checkDefaultPowerShellPath(platformDetails, psExePaths[<powershell name>]);
```

Ideally we'd name PowerShell versions by the version number and architecture (I doubt there's anyone using PowerShell 5.1 Core with VSCode)
```